components/user/Tabs/SettingsContent.tsx

```diff
@@ -1,59 +1,237 @@
-import React from 'react'
+import { useState, useEffect, useCallback } from 'react'
 
-import * as API from 'apiClient'
+import Note from 'components/Form/Note'
+import { FieldError } from 'components/Form/FieldStatus'
 import Select from 'components/Form/Select'
 import TextField from 'components/Form/TextField'
+import Button from 'components/Button'
+import Loader from 'components/Loader'
+// import Debug from 'components/Debug'
+
 import { useField } from 'hooks/useForm'
 import { FIELDS } from 'pages/signup'
 
+import { scrollUp } from 'utils/scroll'
+import { UNSET } from 'utils/forms'
+import { useQueriedToast } from 'hooks/useToast'
+
+import * as API from 'apiClient'
+import { TabType } from './index'
+
 type Props = {
-  authedUser: API.ApiUserMetadata
+  anyBlocksMined: boolean
+  user: API.ApiUser | null
+  authedUser: API.ApiUserMetadata | null
+  toast: ReturnType<typeof useQueriedToast>
+  reloadUser: () => void
+  onTabChange: (tab: TabType) => unknown
 }
 
-const getSocialChoice = (
-  authedUser: API.ApiUserMetadata
-): { choice: string; value: string } => {
-  if (authedUser.discord) {
-    return { choice: 'discord', value: authedUser.discord }
-  } else if (authedUser.telegram) {
-    return { choice: 'telegram', value: authedUser.telegram }
-  }
-  return { choice: '', value: '' }
+const EDITABLE_FIELDS = {
+  email: { ...FIELDS.email, validation: () => true, touched: true },
+  graffiti: { ...FIELDS.graffiti },
+  country: { ...FIELDS.country, useDefault: false },
+  discord: {
+    ...FIELDS.social,
+    required: false,
+    options: undefined,
+    id: 'discord',
+    label: 'Discord',
+    placeholder: 'Your Discord username',
+    isRadioed: false,
+    validation: () => true,
```

Comment:
This doesn't quite match up with the one-or-the-other requirement from signup, are we getting rid of that for now?

Reply:
Ah, yeah. Discussion in Slack starting around [here](https://iron-fish.slack.com/archives/C02PXM6EUDB/p1640037494111500)

Reply:
Hmm, I'm not sure if this totally addresses my question. On the signup page, users are required to either enter a Telegram or enter a Discord username, and they can't leave the field blank (at least one is required). On this settings page, users can delete both fields (neither is required).

One of these changes needs to be made for consistency:
1. This page should validate that one or more of the Telegram or Discord fields has a non-empty value. 
2. OR the signup page should not require a value for Telegram/Discord.

Additionally, our API should not allow empty string/whitespace values to satisfy the field requirement, but I think a frontend change on its own would be good enough for usability.

Reply:
Fair, I will go the second route here.